### PR TITLE
Json cpp support

### PR DESCRIPTION
**Pull request by vovythevov**
This is the first step in the integration of the Parameter Serializer in Slicer.
JsonCpp support is off by default.

### CODE REVIEW

**Comment by fedorov** [comment]
jsoncpp instructions say that "The recommended approach to integrating JsonCpp in your project is to include the amalgamated source" (https://github.com/open-source-parsers/jsoncpp).

What is the advantage of including it as an external project?

File: SuperBuild/External_JsonCpp.cmake [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  if (NOT WIN32)
+    list(APPEND EXTERNAL_PROJECT_OPTIONAL_ARGS
+      -DCMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE}
+      )
+  endif()
+
+  ExternalProject_Add(${proj}
+    ${${proj}_EP_ARGS}
+    GIT_REPOSITORY "${git_protocol}://github.com/KitwareMedical/jsoncpp.git"
+    GIT_TAG "110d054227e9eb63faad48a1fb6a828ad0670e61"
```
Comment (fedorov) [inline SuperBuild/External_JsonCpp.cmake]: what is the difference of this fork and the main location for the project?

https://github.com/open-source-parsers/jsoncpp

**Comment by jcfr** [comment]
> What is the advantage of including it as an external project?

Directly using the amalgamated source means that we would have to explicitly maintain the build system for the jsoncpp library within Slicer or SlicerExcutionModel, since we would like to enable more project to build the library and facilitate maintenance, for now external project sounds reasonable.

> what is the difference of this fork and the main location for the project?

We are currently working on upstreaming few patches.

File: CMakeLists.txt [hunk 2]
```diff
@@ -290,6 +290,9 @@ CMAKE_DEPENDENT_OPTION(
   "Slicer_BUILD_ITKPython" OFF)
 mark_as_superbuild(Slicer_INSTALL_ITKPython)
 
+option(Slicer_BUILD_SERIALIZER "Build Slicer with parameter serializer support" OFF)
```
Comment (jcfr) [inline CMakeLists.txt]: `Slicer_BUILD_PARAMETERSERIALIZER_SUPPORT`

File: SuperBuild.cmake [hunk 3]
```diff
@@ -314,7 +314,7 @@ Slicer_Remote_Add(SlicerDMRI
   GIT_REPOSITORY "${git_protocol}://github.com/SlicerDMRI/SlicerDMRI"
```
Comment (jcfr) [inline SuperBuild.cmake]: If you use `SlicerDMRI/SlicerDMRI@f3a8545` in commit message it will be referenced

Comment (jcfr) [inline SuperBuild.cmake]: `so` -> `yo`

**Comment by jcfr** [comment]
@vovythevov Note that EMSegment has already been updated in [r25293](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=25293)

**Comment by vovythevov** [comment]
Abandoning this PR. It was split in PR#568 and PR#569 for easier integration.

**Pull request closed without merging**